core/.js/src/main/scala/io/udash/utils/FileService.scala

```diff
@@ -0,0 +1,172 @@
+package io.udash.utils
+
+import java.io.{IOException, InputStream}
+
+import org.scalajs.dom._
+import org.scalajs.dom.html.Anchor
+import org.scalajs.dom.raw.Blob
+import scalatags.JsDom
+
+import scala.scalajs.js
+import scala.concurrent.{Future, Promise}
+import scala.scalajs.js.annotation.JSGlobal
+import scala.scalajs.js.typedarray.ArrayBuffer
+import scala.util.Try
+
+@js.native
+@JSGlobal
+sealed class FileReaderSync() extends js.Object {
```

Comment:
I can't ride off from `class` here because it requires to [call constructor](https://w3c.github.io/FileAPI/#FileReaderSync). Like [FileReader](https://w3c.github.io/FileAPI/#APIASynch).

Reply:
I guess it can still be `private final` so we don't ship it in the API.